lisa/schema.py

```diff
@@ -659,35 +683,36 @@ class Environment:
 
     def __post_init__(self, *args: Any, **kwargs: Any) -> None:
         self.nodes: Optional[List[Union[LocalNode, RemoteNode]]] = None
-        if self.nodes_raw is not None:
-            self.nodes = []
-            for node_raw in self.nodes_raw:
-                node_type = node_raw[constants.TYPE]
-                if node_type == constants.ENVIRONMENTS_NODES_LOCAL:
-                    node: Union[
-                        LocalNode, RemoteNode
-                    ] = LocalNode.schema().load(  # type:ignore
-                        node_raw
-                    )
-                    if self.nodes is None:
-                        self.nodes = []
-                    self.nodes.append(node)
-                elif node_type == constants.ENVIRONMENTS_NODES_REMOTE:
-                    node = RemoteNode.schema().load(node_raw)  # type:ignore
-                    if self.nodes is None:
-                        self.nodes = []
-                    self.nodes.append(node)
-                elif node_type == constants.ENVIRONMENTS_NODES_REQUIREMENT:
-                    original_req: NodeSpace = NodeSpace.schema().load(  # type:ignore
-                        node_raw
-                    )
-                    expanded_req = original_req.expand_by_node_count()
-                    if self.nodes_requirement is None:
-                        self.nodes_requirement = []
-                    self.nodes_requirement.extend(expanded_req)
+        if not self.nodes_raw:
```

Comment:
The behavior is changed here. Only `nodes_raw is None`, the `self.nodes is None`. If the `nodes_raws == []`, the `self.nodes == []` also.

Reply:
the `for` statement can work with `[]`, but not `None`. I checked the code, there is no breaks, but I make it always be `[]`, and removed unnecessary protects.

Reply:
Ok, let me see what I can do here.